Services/Dashboard/classes/class.ilDashboardPageLanguageSelectGUI.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+declare(strict_types=1);
+
+use ILIAS\Dashboard\DataRetrieval\Language;
+use ILIAS\Data\URI;
+use ILIAS\DI\Container;
+use ILIAS\UI\URLBuilder;
+
+/**
+ * @@ilCtrl_isCalledBy ilDashboardPageLanguageSelectGUI: ilObjDashboardSettingsGUI
+ */
+class ilDashboardPageLanguageSelectGUI
+{
+    protected Container $dic;
+    private ilDashboardPage $page;
+
+    public function __construct()
+    {
+        global $DIC;
+        $this->dic = $DIC;
+        $this->page = new ilDashboardPage();
+    }
+
+    public function executeCommand(): void
+    {
+        if ($this->dic->ctrl()->getNextClass() === strtolower(ilDashboardPageGUI::class)) {
+            $lang = $this->dic->http()->request()->getQueryParams()[$this->page->getParentType() . '_lang'];
+            if (is_array($lang)) {
+                $lang = reset($lang);
```

Comment:
Why do you need to reset the array. If you want the first element use `current` instead, as `reset` is used for a different purpose.

Reply:
I switch to a proper request wrapper therefore this is no longer needed in any way.